web-server/src/content/Dashboards/ConfigureGithubModalBody.tsx

```diff
@@ -37,25 +39,50 @@ export const ConfigureGithubModalBody: FC<{
     },
     [showError.set]
   );
+  const setDomainError = useCallback(
+    (error: string) => {
+      depFn(showDomainError.set, error);
+    },
+    [showDomainError.set]
+  );
 
   const handleChange = (e: string) => {
     token.set(e);
     showError.set('');
   };
+  const handleDomainChange = (e: string) => {
+    customDomain.set(e);
+    showDomainError.set('');
+  };
+  const checkDomainWithRegex = (domain: string) => {
+    const regex =
+      /^(https?:\/\/)[a-zA-Z0-9]+([-.][a-zA-Z0-9]+)*\.[a-zA-Z]{2,}(:[0-9]{1,5})?(\/.*)?$/;
+    return regex.test(domain);
+  };
```

Comment:
Must include test for this. Very difficult to understand what this regex does. Not sure what the logic is.